packages/test/test-service-load/src/optionsMatrix.ts

```diff
@@ -0,0 +1,135 @@
+import { ISummaryConfiguration } from "@fluidframework/protocol-definitions";
+import { ILoaderOptions  } from "@fluidframework/container-definitions";
+import { IContainerRuntimeOptions, IGCRuntimeOptions, ISummaryRuntimeOptions } from "@fluidframework/container-runtime";
+import { assert } from "@fluidframework/common-utils";
+
+// converts all properties of an object to arrays of the
+// properties potential value. This will be used later
+// to compute a objects that contain pairwise combinations
+// of all property values
+export type OptionsMatrix<T extends Record<string, any>> =
+    Required<{
+        [K in keyof T]: Exclude<T[K], undefined | boolean | number | string> extends  never
+            ? readonly (T[K])[]
+            : readonly (OptionsMatrix<T[K]>)[]
+    }>;
+
+export const booleanCases: readonly (boolean | undefined)[] = [true, false, undefined];
+export const undefinedCases: readonly (undefined)[] = [undefined];
+
+export const loaderOptionsMatrix: OptionsMatrix<ILoaderOptions> = {
+    cache: booleanCases,
+    hotSwapContext: booleanCases,
+    provideScopeLoader: booleanCases,
+    maxClientLeaveWaitTime: undefinedCases,
+    noopCountFrequency: undefinedCases,
+    noopTimeFrequency: undefinedCases,
+};
+
+export const gcOptionsMatrix: OptionsMatrix<IGCRuntimeOptions> = {
+    disableGC: booleanCases,
+    gcAllowed: booleanCases,
+    runFullGC: booleanCases,
+};
+
+export const summaryConfigurationMatrix: OptionsMatrix<Partial<ISummaryConfiguration>> = {
+    idleTime: undefinedCases,
+    maxAckWaitTime: undefinedCases,
+    maxOps: undefinedCases,
+    maxTime: undefinedCases,
+};
+
+export const summaryOptionsMatrix: OptionsMatrix<ISummaryRuntimeOptions> = {
+    disableIsolatedChannels: booleanCases,
+    generateSummaries: booleanCases,
+    initialSummarizerDelayMs: undefinedCases,
+    summaryConfigOverrides:[undefined, summaryConfigurationMatrix],
+};
+
+export const runtimeOptionsMatrix: OptionsMatrix<IContainerRuntimeOptions> = {
+    gcOptions: [undefined, gcOptionsMatrix],
+    summaryOptions: [undefined, summaryOptionsMatrix],
+};
+
+export function buildPairwiseOptions<T>(optionsMatrix: OptionsMatrix<T>): T[] {
```

Comment:
this is slow, but deterministic. there are faster non-deterministic algorithms, but we'd need seeding of randoms across runners to make that tenable, as we want to ensure all runner select from the same set.